src/components/Form/Form.tsx

```diff
@@ -65,19 +64,14 @@ export default function Form<T extends FieldValues>({
   const methods = useForm<T>(formOptions);
   const { handleSubmit, reset } = methods;
 
+  const prevDefaultValues = useRef(defaultValues);
+
   useEffect(() => {
-    if (defaultValues) {
-      if (!deepEqual(defaultValues, methods.getValues())) {
-        Object.entries(defaultValues).forEach(([key, value]) => {
-          methods.setValue(key as Path<T>, value, {
-            shouldDirty: true,
-          });
-        });
```

Comment:
i dont know what this code was originally doing but it buggers up a lot of forms by reseting to the default values just as you submit a form...